### PR TITLE
remove Serializable

### PR DESCRIPTION
**Pull request by SimonCropp**
given binary serializer is no longer supported.

and it was never implementd properly in Polly. take `RateLimitRejectedException` for example. the `RetryAfter` property is not added in GetObjectData and not read back in the deserialize constructor

### CODE REVIEW

**Comment by SimonCropp** [comment]
this came about when converting BrokenCircuitException to nullable. since the nullablility of TResult when deserialized must be null

**Review by martincostello — APPROVED** [review]
Nothing against the change in principle, though once merged this is a breaking change so means any future patch releases we may need to do before v8 will need to be developed from a branch.

**Comment by martintmk** [comment]
Hey @martincostello @SimonCropp do you think we are able to merge this into V8? I encountered this when exploring the codebase and investigating moving some exception types to `Polly.Core` (`ExecutionRejectedException`).

Would simplify some of the next steps.

**Comment by martincostello** [comment]
Fine with me as we've already merged various v8 changes into `main`.

**Comment by martincostello** [comment]
Closing due to merge conflicts and age.

**Pull request closed without merging**
